### PR TITLE
RAC-5637 Add odata endpoint for redfish compliance tests

### PR DESCRIPTION
**Pull request by BrentHiggins**
* Add odata redfish api endpoint
* Add odata.js
* Add odata spec test

### CODE REVIEW

**Comment by BrentHiggins** [comment]
@jimturnquist @hohene

**Comment by BrentHiggins** [comment]
@patelb10

File: lib/api/redfish-1.0/odata.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2017, EMC, Inc.
+
+'use strict';
+
+var injector = require('../../../index.js').injector;
+var controller = injector.get('Http.Services.Swagger').controller;
+var getOdata = controller(function() {
+    var data = {
+        "@odata.context": "/redfish/v1/$metadata",
```
Comment (hohene) [inline lib/api/redfish-1.0/odata.js]: this references $metadata, should it be 'odata'?
seem to work ok with either value when I tried it.

File: spec/lib/api/redfish-1.0/odata-spec.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    afterEach('tear down mocks', function () {
+    });
+
+    after('stop HTTP server', function () {
+        return helper.stopServer();
+    });
+
+    it('should return the correct odata', function () {
+        return helper.request().get('/redfish/v1/$metadata')
```
Comment (BrentHiggins) [inline spec/lib/api/redfish-1.0/odata-spec.js]: Fixed in new push

File: lib/api/redfish-1.0/odata.js (hunk 1 above)
Comment (BrentHiggins) [inline lib/api/redfish-1.0/odata.js]: following redfish spec for odata http://redfish.dmtf.org/schemas/DSP0266_1.1.html#odata-service-document

**Pull request merged**